Issue#54: News CRUD

isvyry opened the pull request:
## ZenHab

* [Main ZenHab ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009/issues/ita-social-projects/home/54)


## Code reviewers

- [x] @alexorlenko
- [x] @Arthurpanda  
- [x] @mykytam 
- [x] @omelchenkotimur 
- [ ] @Prosperro-de 
- [x] @reeztl 

### Second Level Review

- [ ] @DzigMS 
- [ ] @kh0ma 

## Summary of issue

Add News' CRUD operations

## Summary of change

As user I want to view news
As admin I want to be provided with all existing news operations

## Testing approach

---

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

isvyry committed:
Issue#54: News CRUD
Add News entity-stub
Add News repo
Add initial services

isvyry committed:
Merge remote-tracking branch 'origin/Issue#34-Create-User-entity-and-configure-Auth-problem' into Issue#54-News-CRUD

isvyry committed:
Issue#54: News CRUD
Add initial news entity, repository, service, api-service
Edited news section in the spec file

isvyry committed:
Issue#54: News CRUD
Add news service dtos
Add both implementations for service and api-service
Add dependency on generated server to home-application module
Change openapi specification(fix naming, set id to long instead of int)

isvyry committed:
Issue#54: News CRUD
Add news service dtos
Change dto's naming in services

isvyry committed:
Issue#54: News CRUD
Delete unused classes

isvyry committed:
Issue#54: News CRUD
Add custom exceptions
Add paging and sorting to getAll() method

isvyry committed:
Issue#54: News CRUD
Change package naming

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java:
```diff
@@ -0,0 +1,108 @@
+    public Response updateNews(Long id, UpdateNews updateNews, SecurityContext securityContext) {
+        CreateOrUpdateNewsDto updateNewsDto = CreateOrUpdateNewsDto.builder()
+                .title(updateNews.getTitle())
+                .dateTime(LocalDateTime.now())
```
???

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java:
```diff
@@ -0,0 +1,108 @@
+                .text(createNews.getText())
+                .photoUrl(createNews.getPhotoUrl())
+                .source(createNews.getSource())
+                .dateTime(LocalDateTime.now())
```
???

isvyry replied:
What exactly should be fixed?

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java:
```diff
@@ -0,0 +1,108 @@
+    private ReadNews convertToReadNews(ReadNewsDto readNewsDto) {
+        ReadNews toResponse = new ReadNews();
+        toResponse.setId(readNewsDto.getId());
+        toResponse.setCreateOrUpdateDate(OffsetDateTime.of(readNewsDto.getCreatedAt(), ZoneOffset.UTC));
```
Please move the timezone to constant.

isvyry replied:
ok, noted, but what package should constants be in?

kh0ma commented on home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/services/impl/NewsServiceImpl.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    @Override
+    public Collection<ReadNewsDto> getAll(Integer pageNumber, Integer pageSize) {
```
I think we need to return a page but not a collection.

isvyry replied:
so the service layer should return Page, and the api-service provides managing of these pages?

isvyry committed:
Issue#54: News CRUD
Change package naming from plural to singular
Moved UTC Timezone to constants
Moved setting createDate and updateDate to business layer
Set return type to Page in getAll() service method

mykytam reviewed (approved):
Seems everything is good 👍

isvyry committed:
Issue#54: News CRUD
Delete BaseNewsDto

isvyry committed:
Merge remote-tracking branch 'origin/dev' into Issue#54-News-CRUD

isvyry committed:
Issue#54: News CRUD
Rename ApiException to HomeException
Split CreateOrUpdateNewsDto into CreateNewsDto and UpdateNewsDto

isvyry committed:
Issue#54: News CRUD
Service DTOs merged into one general NewsDto
Delete Timezone constant in case we don't need it anymore
Remove create date and update from presentation(edit specification file) and service dtos
